packages/@o3r/testing/src/kassette/update-dates-in-mocks.ts

```diff
@@ -0,0 +1,91 @@
+import type {IMock} from '@amadeus-it-group/kassette';
+import {Temporal} from 'temporal-polyfill';
+
+export interface UpdateDatesInMocksOptions {
+  /**
+   * 'any': dates in inputs will be ignored when computing checksum, the dates on the output will be updated to keep the same day offset
+   * 'day-offset': dates in inputs will be converted to day offsets, the dates on the output will be updated to keep the same day offset
+   * 'same-day-of-week': dates in inputs will be converted to day of week, the dates on the output will be updated to keep the same day of week but increment the week
+   */
+  mode: 'any' | 'day-offset' | 'same-day-of-week';
+  /**
+   * RegExp used to extract dates from request and response
+   */
+  extractor: RegExp;
+  /**
+   * Functions to be used to convert from Temporal objects to strings, and strings to temporal objects
+   */
+  converter: {
+    fromDate: (input: Temporal.ZonedDateTime | Temporal.PlainDate) => string;
+    toDate: (input: string) => Temporal.ZonedDateTime | Temporal.PlainDate;
+  };
+}
+
+/**
+ * Update mock checksum and response body to keep using the same mocks every day with updated dates
+ *
+ * @param mock the mock instance provided by the hook method of Kassette
+ * @param inputOptions default options will extract ISO strings and use 'day-offset' mode
+ */
+export async function updateDatesInMocks(mock: IMock, inputOptions: Partial<UpdateDatesInMocksOptions> = {}) {
+  const options: UpdateDatesInMocksOptions = {
+    mode: 'day-offset',
+    extractor: /\b(\d{4}-\d{2}-\d{2})[^"]*/g,
+    converter: {
+      fromDate: (date) => date.toString().substring(0, 19),
+      toDate: (input) => {
+        input = input.replace(/\.\d+/, '').replace(/Z/, '+00:00');
+        return input.length > 19 ?
+          Temporal.ZonedDateTime.from(`${input.replace(/(\.\d+)/, '')}[${input.replace(/^.*(\+\d{2}:\d{2}).*$/, '$1') || '+00:00'}]`) :
+          Temporal.PlainDate.from(input);
+      }
+    },
+    ...inputOptions
+  };
+  const todayTime = Temporal.Now.plainDateISO();
+
+  // Update request
+  const replaceDatesInInput = (input: string): string => {
+    if (options.mode === 'any') {
```

Comment:
A switch case would help to have a minimum check that you cover all the cases

Reply:
done